pkg/authorization/middleware.go

```diff
@@ -59,6 +60,11 @@ func (m *Middleware) WrapHandler(handler func(ctx context.Context, w http.Respon
 
 		if err := authCtx.AuthZRequest(w, r); err != nil {
 			logrus.Errorf("AuthZRequest for %s %s returned error: %s", r.Method, r.RequestURI, err)
+			if strings.Contains(err.Error(), ErrInvalidPlugin.Error()) {
```

Comment:
IOW, every new authxCtx is populated from middleware's pluginList and this list needs to be updated to prevent further requests from retrying invalid plugin.